src/sequence_state.cc

```diff
@@ -79,30 +84,73 @@ SequenceState::SetStringDataToZero()
         "sequence state to zero.");
   }
 
-  TRITONSERVER_MemoryType memory_type;
-  int64_t memory_type_id;
-
   const std::shared_ptr<MutableMemory>& memory =
       reinterpret_cast<const std::shared_ptr<MutableMemory>&>(Data());
-  char* buffer = memory->MutableBuffer(&memory_type, &memory_type_id);
-  memset(buffer, 0, Data()->TotalByteSize());
+  RETURN_IF_ERROR(memory->SetMemory(0 /* value */));
 
   return Status::Success;
 }
 
+Status
+SequenceState::Resize(
+    void** buffer, const uint64_t buffer_byte_size,
+    TRITONSERVER_MemoryType* memory_type, int64_t* memory_type_id)
+{
+  if (UseGrowableMemory()) {
+    GrowableMemory* growable_memory =
+        reinterpret_cast<GrowableMemory*>(Data().get());
+    RETURN_IF_ERROR(growable_memory->Resize(buffer_byte_size));
+    *buffer = growable_memory->MutableBuffer(memory_type, memory_type_id);
+  } else {
+    std::shared_ptr<AllocatedMemory> memory = std::make_shared<AllocatedMemory>(
+        buffer_byte_size, *memory_type, *memory_type_id);
+    *buffer = memory->MutableBuffer(memory_type, memory_type_id);
+    RemoveAllData();
+    RETURN_IF_ERROR(SetData(memory));
+    if (UseSingleBuffer()) {
+      OtherState()->RemoveAllData();
```

Comment:
for growable memory we are able to change the underlying memory for both input and output in one step transparently - could we do the same for allocated memory? Is there a way to have the two states input / output share a pointer to the same underlying object and hide that detail?

Reply:
will be addressed as a part of memory refactors.